bsmetadata/metadata_utils.py

```diff
@@ -126,20 +164,28 @@ def create_global_metadata_prefix(example: Dict[str, Any], cfg: MetadataConfig)
         cfg: The data config to use.
 
     Returns:
-        A string containing the global metadata prefix.
+        A tuple of two elements, where:
+            - the first element a string containing the global metadata prefix;
+            - the second element is a string contening the name of the metadata types added.
     """
     processed_metadata = {}
+    global_metadata_special_tokens = []
     for metadata in example["metadata"]:
         key, type_ = metadata["key"], metadata["type"]
         if type_ != "global" or key not in cfg.metadata_list:
             continue
 
+        if key not in global_metadata_special_tokens:
+            global_metadata_special_tokens.append(key)
+
```

Comment:
IIUC, this `if` will make the new prefix only contain local metadata types, right?